core/state_transition.go

```diff
@@ -482,9 +478,14 @@ func (st *StateTransition) TransitionDb() (*ExecutionResult, error) {
 	if st.msg.FeeCurrency() == nil && st.msg.MaxFeeInFeeCurrency() != nil {
 		return nil, ErrDenominatedNoCurrency
 	}
-	feeCurrencyRate, err := currency.GetExchangeRate(st.vmRunner, st.msg.FeeCurrency())
-	if err != nil {
-		return nil, err
+	var denominatedCurrencyRate *currency.ExchangeRate
+	// Get rate only for CELO denominated txs (H fork)
+	if st.msg.MaxFeeInFeeCurrency() != nil {
+		var err error
+		denominatedCurrencyRate, err = currency.GetExchangeRate(st.vmRunner, st.msg.FeeCurrency())
+		if err != nil {
+			return nil, err
+		}
 	}
```

Comment:
So this is the actual fix right? Only setting denominatedCurrencyRate for celo denominated transactions.

Reply:
Correct. The rest of the changes are: variable renames for (in my opinion) more explicit names, and changing a few ifblocks to be exactly like they were before the previous change (a mini revert if you will).